dir_to_study/transform.py

```diff
@@ -0,0 +1,153 @@
+    transform(project_id, input_path, output_path, pattern)
+
+
+def transform(project_id, input_path, output_path, pattern):
+    """Transform ResearchStudy, DocumentReference from matching files in input path."""
+    # print(project_id, path, pattern)
+    input_path = pathlib.Path(input_path)
+    output_path = pathlib.Path(output_path)
+
+    for _ in [input_path]:
+        assert _.exists(), f"input_path {_} does not exist."
+        assert _.is_dir(), f"input_path {_} is not a directory."
+
+    for _ in [output_path]:
+        if not _.exists():
+            print(f"output_path {_} does not exist, creating...")
+            _.mkdir(parents=True, exist_ok=True)
+
+    _magic = magic.Magic(mime=True, uncompress=True)
+    program, project = project_id.split('-')
+    research_study = create_research_study(project, f"A study with files from {input_path}/{pattern}")
+
+    emitter(output_path=output_path, name='ResearchStudy').write(
+        orjson.dumps(research_study, orjson.OPT_APPEND_NEWLINE))
+
+    for file in input_path.glob(pattern):
+        if file.is_dir():
+            continue
+        stat = file.stat()
+        modified = datetime.fromtimestamp(stat.st_mtime, tz=timezone.utc)
+        # print(file, stat.st_size, modified.isoformat(), _magic.from_file(file))
+        mime, encoding = mimetypes.guess_type(file)
+        if not mime:
+            mime = _magic.from_file(file)
+
+        _extract_fhir_resources(file)
+
+        document_reference = {
+          "resourceType": "DocumentReference",
+          "status": "current",
+          "docStatus": "final",
+          "id": str(uuid.uuid5(ACED_NAMESPACE, research_study['id'] + "::" + file.name)),
+          "date": modified.isoformat(),  # When this document reference was created
+          "content": [{
+            "attachment": {
+                "extension": [{
+                    "url": "http://aced-idp.org/fhir/StructureDefinition/md5",
+                    "valueString": md5sum(file)
+                }],
+                "contentType": mime,  # Mime type of the content, with charset etc.
+                "url": f"file:///{file}",  # Uri where the data can be found
+                "size": stat.st_size,  # Number of bytes of content (if url provided)
+                "title": file.name,  # Label to display in place of the data
+                "creation": modified.isoformat()  #  Date attachment was first created
+            },
+          }],
+          "context": {  # Clinical context of document
+            "related": [{
+                "reference": f"ResearchStudy/{research_study['id']}"
+            }]  # Related identifiers or resources
+          }
+        }
+        fp = emitter(output_path=output_path, name='DocumentReference')
+        fp.write(orjson.dumps(document_reference, option=orjson.OPT_APPEND_NEWLINE))
+
+    for _ in emitters.values():
+        print(f"wrote {_.name}")
+        _.close()
+
+
+if __name__ == '__main__':
+    transform()
```

Comment:
Refer to the comment above I changed transform() to _transform()

Reply:
Thanks.  Resolved below.